lib/m2ts/m2ts.js

```diff
@@ -343,84 +390,101 @@ ElementaryStream = function() {
 
       stream.size = 0;
 
-      self.trigger('data', event);
+      self.triggerData_(event);
     },
     self;
 
   ElementaryStream.prototype.init.call(this);
   self = this;
 
+  this.triggerData_ = function(data) {
+    sinceFlushDataEmitted++;
+    totalDataEmitted++;
+    this.trigger('data', data);
+  };
+
   this.push = function(data) {
-    ({
-      pat: function() {
-        // we have to wait for the PMT to arrive as well before we
-        // have any meaningful metadata
-      },
-      pes: function() {
-        var stream, streamType;
-
-        switch (data.streamType) {
-        case StreamTypes.H264_STREAM_TYPE:
-        case m2tsStreamTypes.H264_STREAM_TYPE:
-          stream = video;
-          streamType = 'video';
-          break;
-        case StreamTypes.ADTS_STREAM_TYPE:
-          stream = audio;
-          streamType = 'audio';
-          break;
-        case StreamTypes.METADATA_STREAM_TYPE:
-          stream = timedMetadata;
-          streamType = 'timed-metadata';
-          break;
-        default:
-          // ignore unknown stream types
-          return;
-        }
+    if (typeof data !== 'object' ||
+       !data.type) {
+      this.error('Data with no type passed to ElementaryStream');
+      return;
+    }
+    if (data.type === 'pat') {
```

Comment:
I'm not against re-organizing out of the object-property-dispatch here but if we're going to do that, I'd want to split these handlers out into helper functions as well. I think a big chain of if-elses is less readable than switching on well-defined tokens, though.